File: IpAddressJava/IpValidator.java
public class IpValidator {
    
    public static boolean isValidDecimal(String input) {
        if (input == null || input.isEmpty()) {
            return false;
        }
        
        String[] octets = input.split("\\.");
        if (octets.length != 4) {
            return false;
        }
        
        for (String octet : octets) {
            try {
                int value = Integer.parseInt(octet);
                if (value < 0 || value > 255) {
                    return false;
                }
            } catch (NumberFormatException e) {
                return false;
            }
        }
        return true;
    }
    
    public static boolean isValidBinary(String input) {
        if (input == null || input.length() != 32) {
            return false;
        }
        
        for (char c : input.toCharArray()) {
            if (c != '0' && c != '1') {
                return false;
            }
        }
        return true;
    }
    
    public static boolean isValidHexadecimal(String input) {
        if (input == null || input.length() != 8) {
            return false;
        }
        
        for (char c : input.toCharArray()) {
            if (!((c >= '0' && c <= '9') || 
                  (c >= 'A' && c <= 'F') || 
                  (c >= 'a' && c <= 'f'))) {
                return false;
            }
        }
        return true;
    }
}

File: IpAddressPython/ip_validator.py
def is_valid_decimal(input_str):
    if not input_str:
        return False
    
    octets = input_str.split('.')
    if len(octets) != 4:
        return False
    
    for octet in octets:
        try:
            value = int(octet)
            if value < 0 or value > 255:
                return False
        except ValueError:
            return False
    return True

def is_valid_binary(input_str):
    if not input_str or len(input_str) != 32:
        return False
    
    for c in input_str:
        if c not in ['0', '1']:
            return False
    return True

def is_valid_hexadecimal(input_str):
    if not input_str or len(input_str) != 8:
        return False
    
    for c in input_str.lower():
        if c not in '0123456789abcdef':
            return False
    return True
